domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -88,6 +89,50 @@ public async Task<ActionResult<TransferAgreementsResponse>> GetTransferAgreement
         return Ok(new TransferAgreementsResponse(listResponse));
     }
 
+    [ProducesResponseType(204)]
+    [ProducesResponseType(400)]
+    [HttpPatch("{id}")]
+    public async Task<ActionResult<EditTransferAgreementEndDate>> EditEndDate(Guid id, [FromBody] EditTransferAgreementEndDate request)
+    {
+        var validator = new EditTransferAgreementEndDateValidator();
+
+        var validateResult = await validator.ValidateAsync(request);
+        if (!validateResult.IsValid)
+        {
+            validateResult.AddToModelState(ModelState);
+            return ValidationProblem(ModelState);
+        }
+
+        var endDate = DateTimeOffset.FromUnixTimeSeconds(request.EndDate);
+        var senderId = Guid.Parse(User.FindSubjectGuidClaim());
+        var transferAgreement = await transferAgreementRepository.GetTransferAgreement(id);
+
+        if (transferAgreement == null || transferAgreement.SenderId != senderId)
+        {
+            return NotFound();
```

Comment:
Please have a look at the contract service on how we handle it over there.

```
        return result switch
        {
            NonExistingContract => NotFound($"No contract with id {endContract.ContractId} found"),
            MeteringPointOwnerNoMatch => Forbid(),
            Ended => Ok(),
            _ => throw new NotImplementedException($"{result.GetType()} not handled by {nameof(ContractsController)}")
        };
        ```

https://github.com/Energinet-DataHub/energy-origin/blob/main/domains/certificates/Query.API/API/Query.API/Controllers/ContractsController.cs